README.md

```diff
@@ -1,20 +1,108 @@
-# Template project for LF applications based on RIOT OS
+This is a template for Lingua Franca applications targeting RIoT OS on embedded boards. It is currently configured to support ARM-based boards.
 
-This is a template for Lingua Franca applications targeting RIOT OS.
+Outline of these setup instructions:
 
-## Getting started
+[toc]
 
-1. Configure the name of the LF application you want to build inside the `Makefile` by specifying the `LF_MAIN` variable.
-2. Configure the board to flash in the `Makefile` by specifying the `BOARD` variable.
+If you have trouble, please refer to the platform-specific hints above.
 
-### Example
+## 1. Basic Prerequisites
+
+Your system must have the following software packages (you likely have at least some of these already):
+
+ - `git` — [a distributed version control system](https://git-scm.com/)
+ - `make` — Need at least version 4.0 for RIoT (see [macOS Hints](#macos-hints))
+ - `java` — [Java 17](https://openjdk.org/projects/jdk/17)
+ - `nix` — [a purely functional package manager](https://nix.dev/tutorials/install-nix)
+
+**Installation on Linux**
+
+```bash
+sudo apt update
+sudo apt install git openjdk-17-jdk openjdk-17-jre nix cmake
+```
+
+**Installation on macOS**
+
+```bash
+brew install git cmake openjdk@17 make
+curl -L https://nixos.org/nix/install | sh
+```
+
+Note that `make` will be installed as `gmake` (see [macOS Hints](#macos-hints)), so all commands below should use `gmake` instead of `make`.
+
+## 2. Micro C Target for Lingua Franca
+
+This template uses [reactor-uc](https://github.com/lf-lang/reactor-uc), the "micro C" target for Lingua Franca. Clone that repo using one of the following:
+
+```bash
+git clone https://github.com/lf-lang/reactor-uc.git
+git clone git@github.com:lf-lang/reactor-uc.git
```

Comment:
`https://github.com/lf-lang/reactor-uc.git`

Reply:
and I would add the `--recurse-submodules` submodules flag to both of them, then we can remove the `git submodule --init --recursive` step

Reply:
I added this

Reply:
@erlingrj no? 

it should look like this:

```bash
git clone https://github.com/lf-lang/reactor-uc.git --recurse-submodules
git clone git@github.com:lf-lang/reactor-uc.git --recurse-submodules
```